client/my-sites/plan-features-2023-grid/components/features.tsx

```diff
@@ -82,7 +82,7 @@ const PlanFeatures2023GridFeatures: React.FC< {
 						<PlanFeaturesItem>
 							<span className={ spanClasses } key={ key }>
 								<span className={ itemTitleClasses }>
-									{ isFreePlanAndCustomDomainFeature ? (
+									{ isFreePlanAndCustomDomainFeature && domainName ? (
```

Comment:
Should `domainName` empty check also include the else part of this condition?

https://github.com/Automattic/wp-calypso/blob/907954e4dc25dd51b4e26a95bc07f7c7ed8c3b4a/client/my-sites/plan-features-2023-grid/components/features.tsx#L94-L98

Reply:
I don't think it should. It will never reach that point. This is largely the result of returning early on line 59 above:

```
if ( isFreePlanAndCustomDomainFeature && ! domainName ) {
	return null;
}
```

and following with `isFreePlanAndCustomDomainFeature` and using `domainName` within - it creates a type mismatch. One of the reasons I do not like early returns on complex conditions like this. It confuses the rest of the code.

Would you prefer casting `domainName` to string at point of use? (not sure if I do personally)

Reply:
Oh right I have seen this annoyance as well 😬.

> This is largely the result of returning early on line 59 above:

I did not notice this ☝️ .

> Would you prefer casting domainName to string at point of use? (not sure if I do personally)

This would be less confusing for someone looking at the code later, and a legitimate use of `as` since the typescript compiler doesn't seem to be picking up on the correct type here.

Reply:
yeah, that sounds right. updated